source/RTree.hpp

```diff
@@ -52,28 +66,62 @@ class RTree {
   bool empty() const;
 
   void insert(const Rectangle<N> &box, const ElemType &value);
+
   std::shared_ptr<Node> choose_leaf(const std::shared_ptr<Node> &current_node,
                                     const Rectangle<N> &box,
                                     const ElemType &value);
 
   std::shared_ptr<Node> choose_node(const std::shared_ptr<Node> &current_node,
                                     const Rectangle<N> &box,
-                                    SpatialObject *entry);
+                                    SpatialObject *&entry);
 
   std::shared_ptr<Node> adjust_tree(const std::shared_ptr<Node> &parent,
                                     const std::shared_ptr<Node> &left,
                                     const std::shared_ptr<Node> &right,
                                     SpatialObject *entry);
 
+  // TODO(ADE): (COMPLETE) Implement the details of all this functions
+  void find(const Rectangle<N> &box, std::shared_ptr<Node> &temp, std::vector<ElemType> &elements);
   std::vector<ElemType> &operator[](const Rectangle<N> &box);
   std::vector<ElemType> &at(const Rectangle<N> &box);
   const std::vector<ElemType> &at(const Rectangle<N> &box) const;
   // std::vector<ElemType> kNNValue(const Rectangle<N> &box, size_t k) const;
 
-  // private:
+//private:
   std::shared_ptr<Node> root_pointer_;
+  size_t num_of_elements;
 };
 
+template <size_t N, typename ElemType, size_t M, size_t m>
+void RTree<N, ElemType, M, m>::find(const Rectangle<N> &box, std::shared_ptr<Node> &temp, std::vector<ElemType> &elements) {
+  for (size_t i = size_t(0); i < temp->size; i++) {
+    if (!overlaps((*temp)[i].box, box)) {
+      find(box, (*temp)[i].child_pointer, elements);
+    }
+    if (temp->is_leaf()) {
+      elements.push_back((*temp)[i].identifier);
```

Comment:
Debido a que no hay un `else` siempre llega a este punto, por lo que esta añadiendo todas las hojas del `for loop`